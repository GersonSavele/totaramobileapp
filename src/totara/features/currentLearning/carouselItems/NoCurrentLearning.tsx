/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 *
 */

import React, { useContext } from "react";
import {
  View,
  StyleSheet,
  Image,
  Dimensions,
  Text,
  Linking
} from "react-native";

import { PrimaryButton } from "@totara/components";
import { gutter, resizeByScreenSize, ThemeContext } from "@totara/theme";
import { translate } from "@totara/locale";
import { AuthConsumer } from "@totara/core";

const NoCurrentLearning = () => {
  const [theme] = useContext(ThemeContext);

  return (
    <View style={styles.containerStyle}>
      <Image
        style={styles.imageContainer}
        source={require("@resources/images/no_current_learning/no_current_learning.png")}
      />
      <Text style={[theme.textH2, styles.description]}>
        {translate("current_learning.no_learning_message")}
      </Text>
      <AuthConsumer>
        {(auth) =>
          auth.authContextState.appState &&
          auth.authContextState.appState.host && (
            <PrimaryButton
              onPress={() => {
                Linking.openURL(auth.authContextState.appState!.host);
              }}
              text={translate(
                "additional_actions_modal.auth_model_go_to_browser"
              )}
              icon="external-link-alt"
              style={{ alignSelf: "center" }}
            />
          )
        }
      </AuthConsumer>
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    padding: gutter
  },
  imageContainer: {
    height: Dimensions.get("window").width * 0.5,
    width: Dimensions.get("window").width * 0.7,
    resizeMode: "contain",
    paddingBottom: resizeByScreenSize(32, 32, 48, 48)
  },
  description: {
    alignItems: "flex-start",
    marginVertical: resizeByScreenSize(32, 32, 48, 48)
  }
});

export default NoCurrentLearning;
