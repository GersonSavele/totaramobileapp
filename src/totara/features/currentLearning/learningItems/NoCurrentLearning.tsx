/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import React from "react";
import { View, StyleSheet, Image, Text, Linking, ImageSourcePropType } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { PrimaryButton } from "@totara/components";
import { TotaraTheme } from "@totara/theme/Theme";
import { translate } from "@totara/locale";
import { useSession } from "@totara/core";
import { Images } from "@resources/images";
import { paddings } from "@totara/theme/constants";
import { NAVIGATION } from "@totara/lib/navigation";
import { isEnableFindLearning } from "@totara/lib/tools";
import { config } from "@totara/lib";

type NoCurrentLearningProps = {
  testID?: string;
};

const NoCurrentLearning = ({ testID }: NoCurrentLearningProps) => {
  const { host, core } = useSession();
  const navigation = useNavigation();
  const onPressFindLearning = () => {
    if (isEnableFindLearning(core)) {
      navigation.navigate(NAVIGATION.FIND_LEARNING);
    } else {
      Linking.openURL(config.findLearningUri(host!));
    }
  };
  return (
    <View style={styles.containerStyle} testID={testID}>
      <Image source={Images.noCurrentLearning as ImageSourcePropType} />
      <Text style={styles.noCurrentLearningDescription}>{translate("current_learning.no_learning_message")}</Text>
      <View style={styles.goToBrowserAction}>
        <PrimaryButton
          onPress={onPressFindLearning}
          text={translate("current_learning.find_learning")}
          icon={!isEnableFindLearning(core) ? "external-link-alt" : ""}
          style={{ alignSelf: "center" }}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    alignItems: "center"
  },
  noCurrentLearningDescription: {
    ...TotaraTheme.textHeadline,
    fontWeight: "bold",
    paddingTop: paddings.padding3XL
  },
  goToBrowserAction: {
    paddingTop: paddings.padding3XL
  }
});

export default NoCurrentLearning;
