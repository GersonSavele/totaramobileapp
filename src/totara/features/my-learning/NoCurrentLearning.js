/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author Tharaka Dushmantha <tharaka.dushmantha@totaralearning.com>
 **/

import React, { useContext } from "react";
import { View, StyleSheet, Image, Dimensions, Text, Linking } from "react-native";

import { PrimaryButton } from "@totara/components";
import { gutter, resizeByScreenSize, ThemeContext } from "@totara/theme";
import { translate } from "@totara/locale";
import { AuthConsumer } from "@totara/core";
import ScormActivityView from "@totara/activities/scorm/SCORMActivity"
import {ScormActivity} from "@totara/types"

const NoCurrentLearning = () => {

  const [theme] = useContext(ThemeContext);

  return (
    <View style={styles.containerStyle}>
      <Image style={styles.imageContainer} source={require("@resources/images/no_current_learning/no_current_learning.png")} />
      <Text style={[theme.textH2, styles.description]}>{translate("my-learning.no_learning_message")}</Text>
      <AuthConsumer>
        {auth => (
          <PrimaryButton onPress={() => { Linking.openURL(auth.appState.host); }} text={translate("additional-actions-modal.auth_model_go_to_browser")} icon="external-link-alt" style={{ alignSelf: "center" }} />
        )}
      </AuthConsumer>
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    padding: gutter
  },
  imageContainer: {
    height: Dimensions.get("window").width * 0.5,
    width: Dimensions.get("window").width * 0.7,
    resizeMode: "contain",
    paddingBottom: resizeByScreenSize(32, 32, 48, 48)
  },
  description: {
    alignItems: "flex-start",
    marginVertical: resizeByScreenSize(32, 32, 48, 48)
  }
});

export default NoCurrentLearning;
