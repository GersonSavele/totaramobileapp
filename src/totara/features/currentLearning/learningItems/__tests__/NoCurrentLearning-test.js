/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */
import React from "react";
import renderer from "react-test-renderer";
import * as ReactRedux from "react-redux";
import NoCurrentLearning from "../NoCurrentLearning";

describe("NoCurrentLearning, No current learning screen should launch when api return doesn't have courses", () => {
  it("Test result : take screen-shot when no current learning", () => {

    jest.spyOn(ReactRedux, "useSelector").mockImplementation(() => {
      return {
        host: 'https://mobile.demo.totara.software'
      };
    });

    const tree = renderer.create(<NoCurrentLearning />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
