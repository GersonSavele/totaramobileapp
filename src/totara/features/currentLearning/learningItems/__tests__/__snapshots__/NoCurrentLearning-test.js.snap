// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoCurrentLearning, No current learning screen should launch when api return doesn't have courses Test result : take screen-shot when no current learning 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Image
    source="test-file-stub"
  />
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 28,
        "paddingTop": 32,
      }
    }
  >
    translated current_learning.no_learning_message with undefined
  </Text>
  <View
    style={
      Object {
        "paddingTop": 32,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          this is a mock button
        </Text>
      </View>
    </View>
  </View>
</View>
`;
